File: examples/plotting/file/categorical.py
from bokeh.layouts import row
from bokeh.plotting import figure, show, output_file

factors = ["a", "b", "c", "d", "e", "f", "g", "h"]
x =  [50, 40, 65, 10, 25, 37, 80, 60]

dot = figure(title="Categorical Dot Plot", tools="", toolbar_location=None,
            y_range=factors, x_range=[0,100])

dot.segment(0, factors, x, factors, line_width=2, line_color="green", )
dot.circle(x, factors, size=15, fill_color="orange", line_color="green", line_width=3, )

factors = ["foo", "bar", "baz"]
x = ["foo", "foo", "foo", "bar", "bar", "bar", "baz", "baz", "baz"]
y = ["foo", "bar", "baz", "foo", "bar", "baz", "foo", "bar", "baz"]
colors = [
    "#0B486B", "#79BD9A", "#CFF09E",
    "#79BD9A", "#0B486B", "#79BD9A",
    "#CFF09E", "#79BD9A", "#0B486B"
]

hm = figure(title="Categorical Heatmap", tools="hover", toolbar_location=None,
            x_range=factors, y_range=factors)

hm.rect(x, y, color=colors, width=1, height=1)

output_file("categorical.html", title="categorical.py example")

show(row(hm, dot, sizing_mode="scale_width"))  # open a browser

File: bokehjs/examples/categorical/categorical.ts
namespace Categorical {
  import plt = Bokeh.Plotting
  const {Row} = Bokeh

  console.log(`Bokeh ${Bokeh.version}`)
  Bokeh.set_log_level("info")
  Bokeh.settings.dev = true

  const dot = () => {
    const factors = ["a", "b", "c", "d", "e", "f", "g", "h"]
    const x =  [50, 40, 65, 10, 25, 37, 80, 60]

    const fig = plt.figure({title: "Categorical Dot Plot", tools: "", toolbar_location: null, y_range: factors, x_range: [0, 100]})

    fig.segment(0, factors, x, factors, {line_width: 2, line_color: "green"})
    fig.circle(x, factors, {size: 15, fill_color: "orange", line_color: "green", line_width: 3})
    return fig
  }

  const hm = () => {
    const factors = ["foo", "bar", "baz"]
    const x = ["foo", "foo", "foo", "bar", "bar", "bar", "baz", "baz", "baz"]
    const y = ["foo", "bar", "baz", "foo", "bar", "baz", "foo", "bar", "baz"]
    const colors = [
      "#0B486B", "#79BD9A", "#CFF09E",
      "#79BD9A", "#0B486B", "#79BD9A",
      "#CFF09E", "#79BD9A", "#0B486B",
    ]

    const fig = plt.figure({title: "Categorical Heatmap", tools: "hover", toolbar_location: null, x_range: factors, y_range: factors})

    fig.rect(x, y, 1, 1, {color: colors})
    return fig
  }

  plt.show(new Row({children: [hm(), dot()], sizing_mode: "scale_width"}))
}
